### PR TITLE
Offchain2

### PR DESCRIPTION
**Pull request by makoto**
- Replace the internal of `getResolver` to call `provider.resolver(name)` instead of calling contract function directly.
- Add `wildcardResolverDomain` function
- Pass ensAddress if provider is localhost with chainId as 1337

NOTE: There are several functions not supporting CCIP / wildcard due to the lack of ethers.js Should have no impact as these functions are not used within `ens-app`.

### CODE REVIEW

File: src/ens.js [hunk 1]
```diff
@@ -79,77 +78,59 @@ export class ENS {
...
   /* Main methods */
 
+  // TODO: ethers.js does not support owner
   async getOwner(name) {
     const namehash = getNamehash(name)
     const owner = await this.ENS.owner(namehash)
     return owner
   }
 
   async getResolver(name) {
```
Comment (Arachnid) [inline src/ens.js:88]: This should probably be called `getResolverAddress`, and `getResolverObject` should be renamed to `getResolver`. This function can then call that one and just unwrap the resulting object.

File: src/ens.js [hunk 2]
```diff
@@ -556,7 +498,12 @@ export class ENS {
     let namehash = getNamehash(reverseNode)
     return Resolver.setName(namehash, name)
   }
-
+  async wildcardResolverDomain(name){
```
Comment (Arachnid) [inline src/ens.js:501]: This definitely needs a different name - something like 'hasResolver'?

**Review by Arachnid — CHANGES_REQUESTED** [review]

File: src/ens.js [hunk 3]
```diff
@@ -556,7 +500,21 @@ export class ENS {
...
+    let res, resNew
+    const provider = await getProvider()
+    const resolverAddress = await this.getResolver(name)
+    const _resolverAddress = await provider._getResolver(name)
```
Comment (Arachnid) [inline src/ens.js:507]: ??? what is going on here?

File: src/constants/interfaces.js [hunk 4]
```diff
@@ -6,5 +6,9 @@ export const interfaces = {
   dnsRegistrar: '0x1aa2e641',
   bulkRenewal: '0x3150bfba',
   dnssecClaimOld:'0x1aa2e641',
-  dnssecClaimNew:'0x17d8f49b'
+  dnssecClaimNew:'0x17d8f49b',
+  // new interface id https://docs.ens.domains/ens-improvement-proposals/ensip-10-wildcard-resolution#specification
+  resolveNew:'0x9061b923',
+  // previous interface id https://github.com/ensdomains/ens-contracts/blame/ed3652275e71e8811c9ee690a14a6492f461a6d4/contracts/resolvers/profiles/NameResolver.sol#L5
```
Comment (Arachnid) [inline src/constants/interfaces.js:12]: This is the interface ID for the 'name' function, not for 'resolve'.

File: src/ens.js (hunk 2 above)
Comment (Arachnid) [reply src/ens.js:503]: This should at least be 'hasWildcardResolver' or somesuch; the current name implies it returns a 'domain', not a bool.

**Review by Arachnid — CHANGES_REQUESTED** [review]

File: src/ens.js (hunk 1 above)
Comment (makoto) [reply src/ens.js:90]: `getResolver` is a public function so I don't want to change the interface and also shouldn't return the entire ethersjs's resolver object unless necessary (our library interface changes whenever ethers.js interface changes). I can  rename `getResolverObject` to `_getResolverObject` to make it look more like private function if necessarily.

File: src/ens.js (hunk 3 above)
Comment (makoto) [reply src/ens.js:507]: _getResolvere calls registry.resolver directly bypassing wildcard. I initially added it to address
https://github.com/ensdomains/ens-app/pull/1468#discussion_r854544730  (`We should show that message for any domain which a) doesn't exist in the registry, or b) has a resolver that implements resolve`) but if I think about it we should just check if the resolver implements `resolve` (my original implementation).

**Pull request merged**